File: src/main/domain/folders/FolderRepository.ts
import Folder from "./Folder";
import FoldersFilter from "./FoldersFilter";

export default interface FolderRepository {
  save(folder: Folder): Promise<Folder>
  find(filter: FoldersFilter): Promise<Folder[]>;
  findById(id: number): Promise<Folder | null>;
  delete(id: number): Promise<boolean>;
}

File: src/domain/cards/Folder.Repository.go
package cards

type FolderRepository interface {
	Insert(folder *Folder) (*Folder, error)
	Update(folder *Folder) (*Folder, error)
	Find(filter FolderFilter) (*[]*Folder, error)
	FindById(id *int) (*Folder, error)
	FindByParentId(id *int) (*[]*Folder, error)
	Delete(folderId *int) (bool, error)
}
